### DISCUSSION TITLE
mantel test

### DISCUSSION THREAD

**Discussion by EvoArt** [EcoJulia/Microbiome.jl#128] [General]
see bottom of here for a rough and ready mantel test implementation https://github.com/EvoArt/Diversity.jl/blob/dev/src/Testing/permutation_tests.jl There's lot's more can be added (two tailed test, rank correlation, partial mantel test). 

I'm not sure where all this kind of code should be living though. The kind of stats community ecologists tend to do, should we be keeping them under Diversity.jl (or another community ecology package) to be looked after by ecologists, or should it all be in MultivariateStats (or even StatsBase)? Maybe the latter is more sensible.

**Comment by mkborregaard** [comment]
Maybe https://github.com/EcoJulia/RandomBooleanMatrices.jl could be expanded to contain stuff like this too?

**Reply by kescobo** [reply]
That package seems designed to do a single thing well. If the functionality is useful for other things, I think better to use it as a dependency rather than adding bloat to it, no?

**Reply by mkborregaard** [reply]
That's a good point. It would change the scope of the package for sure, I agree, but that might be OK. I also like your idea of an EcoStats package, though in my view most of what we do is eco-stats, so it's a bit hard for me to circumscribe the scope for that?

**Reply by kescobo** [reply]
> most of what we do is eco-stats, so it's a bit hard for me to circumscribe the scope for that?

Definitely. I was thinking `EcoStats.jl` might just be a bunch of `@reexport` statements or something :-).

**Reply by mkborregaard** [reply]
Hmm then I'm not really in favour, that seems to go against having small packages that do defined things too

**Reply by kescobo** [reply]
Well, I think there's _also_ room for omnibus packages that make things a bit easier on users by aggregating functionality and preventing them from having 20 `using` statements at the top of an analysis script. But this is uncommon in julia, I admit.

**Comment by kescobo** [comment]
> I'm not sure where all this kind of code should be living though. The kind of stats community ecologists tend to do, should we be keeping them under Diversity.jl (or another community ecology package) to be looked after by ecologists, or should it all be in MultivariateStats (or even StatsBase)? Maybe the latter is more sensible.

Or maybe `HypothesisTests.jl`

Another thought would be to make an `EcoStats.jl` package, but in general, I'm most in favor of having small packages that do defined things. At first, I developed a bunch of stuff for myself in `Microbiome.jl`, but I've been trying to move as much as possible out of it to other relevant packages, and then depending on those packages. Eg, I initially had a within-package Bray Curtis dissimilarity function, but decided to instead put it into Distances.jl, and then depend on that package.

So here's another thought: pull out anything that could be used for both mantel testing and PERMANOVA into a `PermutationTestsBase.jl` or something, then have `PERMANOVA.jl` and `Mantel.jl` depend on that. At some point, we might want to make `EcoStats.jl` package that uses / reexports those other packages, but

**Comment by kescobo** [comment]
@EvoArt Hope you don't mind, I converted this to a discussion instead of an issue
